### ISSUE TITLE
Serve transit data through our own API

### ISSUE THREAD

**Issue by jeancochrane** [jeancochrane/septa_tracker#2]
Something like [this](https://blog.miguelgrinberg.com/post/designing-a-restful-api-with-python-and-flask). We need to:
1. Retrieve SEPTA real-time data from our back-end
2. Clean it; test for mislabelled and unlabelled vehicles, as well as vehicles that are off their routes
3. Serve data through an API

**Comment by kalilsn** [comment]
I think we should have one process running that periodically (tbd how often) retrieves data from [here](http://www3.septa.org/hackathon/TransitViewAll/), then does our cleaning, and stores it in an SQL database. Then the api will mostly just be sql queries – and as long as our database is designed to be pretty universal, we should be able to reimplement this for other cities without altering anything but the process that is pulling data from SEPTA/its counterpart.

Ideas for api endpoints:
- [ ] Get bus positions by route
- [ ] Get bus positions by stop
- [ ] Get bus timings by stop
- [ ] Get bus positions by long/lat (maybe unnecessary given the previous one but potentially useful)

**Comment by jeancochrane** [comment]
Agreed! I'd also like to serve fixed routelines as a separate API, too.

Our cleaning script will most likely be the biggest headache here, and might even warrant its own issue thread. For now, let's focus on setting up a mirror API on our server, and then work on cleaning the data after we've shifted our architecture. Here's my suggested task list:
- [ ] 1. Set up automatic querying of TransitViewAll 
- [ ] 2. Build DB for storing the data on our server (Postgres?)
- [ ] 3. Build API in Flask, routing queries with SQLAlchemy 
- [ ] 4. Switch our current app architecture to query from our own API instead of SEPTA's
- [ ] 5. Start on data cleaning script
